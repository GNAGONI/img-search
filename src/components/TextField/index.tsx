import { styled } from '@mui/material/styles';
import TextField from '@mui/material/TextField';
import { SyntheticEvent } from 'react';

interface Props {
  label: string;
  value: string;
  onChange: (e: SyntheticEvent) => void
}

const CustomStyledTextField = styled(TextField)({
  width: '300px',
  margin: '4px',
});


const CustomTextField = (props: Props) => {
  return (
    <CustomStyledTextField
      label={props.label}
      variant="outlined"
      type="text"
      required
      value={props.value}
      onChange={props.onChange}
    />
  );
};

export default CustomTextField;
