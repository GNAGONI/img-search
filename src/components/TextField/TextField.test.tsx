import renderer from 'react-test-renderer';
import TextField from './index';

it('renders correctly TextField', () => {
  const mockCallback = jest.fn();
  const tree = renderer
    .create(
      <TextField label="Test" value="test" onChange={mockCallback} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});