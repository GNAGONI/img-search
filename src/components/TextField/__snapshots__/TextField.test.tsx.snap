// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly TextField 1`] = `
<div
  className="MuiFormControl-root MuiTextField-root css-t7h6ca-MuiFormControl-root-MuiTextField-root"
>
  <label
    className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiFormLabel-filled Mui-required MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined css-1sumxir-MuiFormLabel-root-MuiInputLabel-root"
    data-shrink={true}
    htmlFor=":r0:"
    id=":r0:-label"
  >
    Test
    <span
      aria-hidden={true}
      className="MuiFormLabel-asterisk MuiInputLabel-asterisk css-wgai2y-MuiFormLabel-asterisk"
    >
       
      *
    </span>
  </label>
  <div
    className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-formControl css-9ddj71-MuiInputBase-root-MuiOutlinedInput-root"
    onClick={[Function]}
  >
    <input
      aria-invalid={false}
      autoFocus={false}
      className="MuiInputBase-input MuiOutlinedInput-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
      disabled={false}
      id=":r0:"
      onAnimationStart={[Function]}
      onBlur={[Function]}
      onChange={[Function]}
      onFocus={[Function]}
      required={true}
      type="text"
      value="test"
    />
    <fieldset
      aria-hidden={true}
      className="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
    >
      <legend
        className="css-14lo706"
      >
        <span>
          Test
           
          *
        </span>
      </legend>
    </fieldset>
  </div>
</div>
`;
